oval_graph/client.py

```diff
@@ -242,21 +240,25 @@ def _merge_report_parts(self, data):
             *body,
             *footer]
 
-    def save_html_report(self, dict_, src, status_all_in_one=False):
-        data = ("\n<script>var all_in_one = " +
-                str(status_all_in_one).lower() +
-                ";</script>" +
-                "\n<script>var data_of_tree =" +
-                str(json.dumps(dict_, sort_keys=False, indent=4)) +
-                ";</script>\n")
-        if status_all_in_one:
-            for rule in dict_:
-                data += ('<h1>' +
-                         rule.split('-')[2] +
-                         '</h1>' +
-                         '<div id="' +
-                         re.sub(r'[\_\-\.]', "", rule) +
-                         '"></div>')
+    def save_html_report(self, dict_, src):
+        data = (
+            "\n<script>var data_of_tree = " + str(
+                json.dumps(
+                    {
+                        re.sub(
+                            r'\-\d{2}\_\d{2}\_\d{4}\-\d{2}\_\d{2}\_\d{2}$',
+                            '',
+                            k): v for k,
+                        v in dict_.items()},
+                    sort_keys=False,
+                    indent=4)) + ";</script>\n")
```

Comment:
It is totally unclear what this code is doing. `data` and `dict_` are generic names, and the regular expression is complicated. Consider extracting this functionality into a named private method, and naming the regexp by storing it in a variable.

Aside from that, you are creating contents of an HTML document, so you could even consider using the lxml functionality, and create parts of it programatically.